android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java

```diff
@@ -484,30 +413,21 @@ private void updatePortfolioGetPendingTx() {
                                 mPortfolioHelper.calculateBalances(() -> {
                                     final String fiatSumString = String.format(Locale.getDefault(),
                                             "$%,.2f", mPortfolioHelper.getTotalFiatSum());
-                                    PostTask.runOrPostTask(UiThreadTaskTraits.DEFAULT, () -> {
-                                        mFiatSumString = fiatSumString;
-                                        mBalance.setText(mFiatSumString);
-                                        mBalance.invalidate();
-
-                                        List<BlockchainToken> tokens = new ArrayList<>();
-                                        List<BlockchainToken> nfts = new ArrayList<>();
-
-                                        for (BlockchainToken token :
-                                                mPortfolioHelper.getUserAssets()) {
-                                            if (token.isErc721 || token.isNft) {
-                                                nfts.add(token);
-                                            } else {
-                                                tokens.add(token);
-                                            }
+                                    mFiatSumString = fiatSumString;
+                                    mBalance.setText(mFiatSumString);
+                                    mBalance.invalidate();
+
+                                    List<BlockchainToken> tokens = new ArrayList<>();
+
+                                    for (BlockchainToken token : mPortfolioHelper.getUserAssets()) {
+                                        if (!token.isErc721 && !token.isNft) {
+                                            // Add only coins and not NFTs.
+                                            tokens.add(token);
```

Comment:
nit: not required, but some point we can have separate call (or flags) to filter out NFTs and tokens (maybe in TokenUtils), to avoid fetching and tracking balance of unwanted assets.